.github/workflows/build.yml

```diff
@@ -127,6 +121,22 @@ jobs:
         with:
           gradle-home-cache-cleanup: true
 
+      # Setup Go (including module cache) for license-header
+      - uses: actions/setup-go@v5
+        with:
+          go-version: 'stable'
+          check-latest: true
+
+      - name: Cache Go Modules
```

Comment:
Since there isn't a `go.mod`, the default caching doesn't have any effect. This will use the gradle config as a cache key which ensures we'll rebuild the cache when the buf version changes (or if we change the build steps for any reason).